app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -106,8 +124,8 @@ def stack_response
     @stack_response ||= Terraform::Runner::ResponseAsync.new(options[:terraform_stack_id])
   end
 
-  def decrypt_extra_vars(extra_vars)
-    result = extra_vars.deep_dup
+  def decrypt_input_vars(input_vars)
+    result = input_vars.deep_dup
     result.transform_values! { |val| val.kind_of?(String) ? ManageIQ::Password.try_decrypt(val) : val }
```

Comment:
@putmanoj could you do `input_vars.transform_values { |val| val.kind_of?(String) ? ManageIQ::Password.try_decrypt(val) : val }` instead of `deep_dup` plus `transform_values!` ?

`transform_values!` means you're going out of your way to change the original object not create a copy, but it looks like you _do_ want to create a copy and not modify the original